### PR TITLE
Algo Cash - An algorithmic stable coin

### PR DESCRIPTION
**Pull request by Dohkooo**
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC or Ethereum address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants at the time of submission.
- [x] The initial PR contains only one commit (squash if needed before submitting your PR).
- [x] The grant will only be announced once we successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for your application. This looks quite interesting to me. However the deliveries currently don’t contain the ink! Smart contract and the implementation of the protocol itself. From a grant perspective, we support the development of open source software that is potentially useful for others in our ecosystem. We wouldn’t be interested in supporting a Fornt End unless the purpose of the UI is to interact with the ink! Smart contract or on-chain protocol. Also are you aware of: https://www.youtube.com/watch?v=bnC5RQsaAXQ? This might be helpful for you.

**Comment by Dohkooo** [comment]
> Thanks for your application. This looks quite interesting to me. However the deliveries currently don’t contain the ink! Smart contract and the implementation of the protocol itself. From a grant perspective, we support the development of open source software that is potentially useful for others in our ecosystem. We wouldn’t be interested in supporting a Fornt End unless the purpose of the UI is to interact with the ink! Smart contract or on-chain protocol. Also are you aware of: https://www.youtube.com/watch?v=bnC5RQsaAXQ? This might be helpful for you.

Thanks for your feedback. Do you mean the application would be accepted only after the implementation of the protocol and !ink Smart Contract having been done?

**Comment by Dohkooo** [comment]
Yes, we will also implement a smart contract repo with !ink in MileStone 1. And the whole protocol is also implemented by smart contract.

Thanks for your remind of https://www.youtube.com/watch?v=bnC5RQsaAXQ. It's a very good reference and aligned with our design.

**Comment by Dohkooo** [comment]
Just wondering if you have a preference of the implementation regarding to smart contract vs pallet? Or you think they are both worth?

**Review by Noc2 — CHANGES_REQUESTED** [review]
It’s up to you to decide, which solution you think is better suited for your implementation. You can take a look at this [post](https://stackoverflow.com/questions/56040779/when-should-i-build-a-substrate-runtime-module-versus-a-substrate-smart-contract). Will it be possible to interact with your pallet/smart contract via the UI. If not you can probably remove it from the deliveries.  How is your solution different from the video/pallet, which I shared above?

File: applications/AlgoCash.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+### Milestone 1 Example — Implement Substrate Modules 
+* **Estimated Duration:** 7 weeks
+* **FTE:**  2
+* **Costs:** 5,000 DAI  
+
+| Number | Deliverable | Specification |
+| ------------- | ------------- | ------------- |
+| 0 | License | Apache 2.0 / MIT / Unlicense |
+| 1 | Documentation | Specification of the background, components and working mechanism|
+| 2 | Smart Contract | AlgoCash smart contract repo. The smart contract can be deployed to any substrate chain with ink! pallet.|
```
Comment (Noc2) [inline applications/AlgoCash.md:150]: Could you specify the functions of the smart contract here a little bit better? These are basically the requirements of our contracts.

File: applications/AlgoCash.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+| 3 | Tests |Unit Test and also we will test it on Canvas| 
+| 4 | Fornt End | Project website to distribute ALC, crowd sale ALS and oprate with DEXs/Bond Exchange/Board Room |
```
Comment (Noc2) [inline applications/AlgoCash.md:152]: You can remove any parts related to your token distribution/sale from the deliveries. We are only interested in the open source software that you develop as part of the grant and want to have as much details as possible about it.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. I added a few additional comments below. In general I recommend to focus not too much on the Token Distribution as long as you are still experimenting, which is perfectly fine for grants.

File: applications/AlgoCash.md (hunk 1 above)
Comment (Dohkooo) [reply applications/AlgoCash.md:208]: The specification of smart contract has been added as a section

File: applications/AlgoCash.md (hunk 2 above)
Comment (Dohkooo) [reply applications/AlgoCash.md]: Removed Front End from the deliveries.

**Comment by Noc2** [comment]
Thanks for the updates. Looks good to me and I’m happy to go ahead with it. I will share it with the rest of the team.

**Comment by Dohkooo** [comment]
@Noc2 Thanks for the review and appreciate all your comments.

**Review by Noc2 — APPROVED** [review]

**Review by BenWhiteJam — APPROVED** [review]

**Review by mmagician — APPROVED** [review]

**Comment by alxs** [comment]
@Dohkooo care to share a quick status update on the project, together with an ETA if possible? Your first delivery is delayed by several months so if we don't hear something from you soon, we will assume you're no longer working on this and terminate the grant due to inactivity.

**Comment by Dohkooo** [comment]
Hi @alxs, I'm very sorry for the delay and not having update here. We've met some health issues during the past a few weeks. Now we're working very hard towards our first milestone delivery. We are about to finish the development of all functionalities and going to setup the test environment. ETA is that we will make our milestone delivery by the end of this month. Sorry again for the late reponse.

**Comment by alxs** [comment]
Hope your team is feeling better. Sounds good, looking forward to your delivery.

**Pull request merged**